pippo-core/src/main/java/ro/fortsoft/pippo/core/Response.java

```diff
@@ -388,14 +388,64 @@ public boolean isCommitted() {
         return httpServletResponse.isCommitted();
     }
 
+    /**
+     * Attempts to negotiate the Response content type based on the Request
+     * accept type. If an agreement can't be reached, the Response content-type
+     * is not set.
+     *
+     * This allows a response chain such as
+     * <code>response.xml().negotiate(request).send(myObject);</code> which will
+     * set the content type as <code>application/xml</code> and optionally
+     * override it after negotation.
+     *
+     * @param request
+     * @return the response
+     */
+    public Response negotiate(Request request) {
```

Comment:
To be honest I don't understand the scenario (I have the feeling that it's related to REST API) for a construction like `xml().contentType(request)`. It's not so predictable the final value of contentType.